src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/struckout/StruckOutAboutToSubmitHandler.java

```diff
@@ -0,0 +1,47 @@
+package uk.gov.hmcts.reform.sscs.ccd.presubmit.struckout;
+
+import static java.util.Objects.requireNonNull;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.sscs.ccd.callback.Callback;
+import uk.gov.hmcts.reform.sscs.ccd.callback.CallbackType;
+import uk.gov.hmcts.reform.sscs.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.sscs.ccd.domain.CaseDetails;
+import uk.gov.hmcts.reform.sscs.ccd.domain.EventType;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
+import uk.gov.hmcts.reform.sscs.ccd.presubmit.PreSubmitCallbackHandler;
+
+@Service
+@Slf4j
+public class StruckOutAboutToSubmitHandler implements PreSubmitCallbackHandler<SscsCaseData> {
+
+    @Override
+    public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callback) {
+        requireNonNull(callback, "callback must not be null");
+        requireNonNull(callbackType, "callbacktype must not be null");
+
+        return callbackType.equals(CallbackType.ABOUT_TO_SUBMIT)
+                && callback.getEvent() == EventType.STRUCK_OUT;
+    }
+
+    @Override
+    public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType, Callback<SscsCaseData> callback,
+                                                          String userAuthorisation) {
+        if (!canHandle(callbackType, callback)) {
+            throw new IllegalStateException("Cannot handle callback.");
+        }
+        final CaseDetails<SscsCaseData> caseDetails = callback.getCaseDetails();
+        final SscsCaseData sscsCaseData = caseDetails.getCaseData();
+
+        log.info(String.format("Handling struckOut event for caseId %s", sscsCaseData.getCcdCaseId()));
+
+        sscsCaseData.setInterlocReviewState(null);
+        sscsCaseData.setDirectionDueDate(null);
+        sscsCaseData.setHmctsDwpState("strikeOutActioned");
```

Comment:
Can you please add "strikeOutActioned" as constant variable or state also do we have to set the DwpState here? e.g caseData.setDwpState(DwpState.STRUCK_OUT.getId());